### PR TITLE
Implement HTMLSelectElement.selectedOptions

### PR DESCRIPTION
**Pull request by Zirro**
Implements: https://html.spec.whatwg.org/multipage/forms.html#dom-select-selectedoptions

Includes a to-upstream test as one did not exist in the testsuite.

### CODE REVIEW

File: lib/jsdom/living/nodes/HTMLSelectElement-impl.js [hunk 1]
```diff
@@ -90,6 +90,12 @@ class HTMLSelectElementImpl extends HTMLElementImpl {
     return this.options.length;
   }
 
+  get selectedOptions() {
+    return createHTMLCollection(this, () => domSymbolTree.treeToArray(this, { filter(node) {
```
Comment (Sebmaster) [inline lib/jsdom/living/nodes/HTMLSelectElement-impl.js]: Does `createHTMLCollection` ensure that `[SameObject]` works? I think we might have to cache this object in a private var @domenic?

**Comment by Zirro** [comment]
@domenic Thanks! I pushed a minor change to give the CI another spin. The single failure looks sporadic.

File: lib/jsdom/living/nodes/HTMLSelectElement-impl.js (hunk 1 above)
Comment (domenic) [inline lib/jsdom/living/nodes/HTMLSelectElement-impl.js]: Oh wow, I totally skipped over the [SameObject]. You're right; we should store it in a private var (and probably create it lazily). And add tests for that!!

Comment (Zirro) [inline lib/jsdom/living/nodes/HTMLSelectElement-impl.js]: This is probably beyond my current knowledge about jsdom. I appreciate if you're willing to make this change.

Comment (domenic) [inline lib/jsdom/living/nodes/HTMLSelectElement-impl.js]: Can do, but it's not so bad. Basically just `return this._selectedOptions || (this._selectedOptions = ...);`, plus a test that el.selectedOptions === el.selectedOptions (and maybe some tests that changing selection also preseves identity).

Comment (Zirro) [inline lib/jsdom/living/nodes/HTMLSelectElement-impl.js]: Oh, I see. If you don't mind waiting until the weekend, I can give it a try.

**Comment by Zirro** [comment]
The [SameObject] support will be resolved with https://github.com/jsdom/webidl2js/pull/37, and I've added a test which should pass with those changes.

File: test/web-platform-tests/to-upstream/html/semantics/forms/the-select-element/select-selectedOptions.html [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+</select>
+
+<script>
+"use strict";
+
+test(() => {
+  const select = document.getElementById("select-none-selected");
+
+  assert_equals(select.selectedOptions.item(0), select.children[0]);
+  assert_equals(select.selectedOptions.length, 1);
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/html/semantics/forms/the-select-element/select-selectedOptions.html]: I am pretty sure you can make this more idiomatic by using `assert_array_equals(select.selectedOptions, [select.children[0]]);`

File: lib/jsdom/living/nodes/HTMLSelectElement-impl.js (hunk 1 above)
Comment (domenic) [inline lib/jsdom/living/nodes/HTMLSelectElement-impl.js]: Can you update the package.json for webidl2js v6.1.0? You may need to rebase.

File: test/web-platform-tests/to-upstream/html/semantics/forms/the-select-element/select-selectedOptions.html [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+}, ".selectedOptions without the 'multiple' attribute but " +
+   "more than one selected option should return the last one");
+
+test(() => {
+  const select = document.getElementById("select-same-object");
+  const selectAgain = document.getElementById("select-same-object");
+
+  assert_equals(select.selectedOptions, selectAgain.selectedOptions);
+
+}, ".selectedOptions should always return the same value - [SameObject]");
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/html/semantics/forms/the-select-element/select-selectedOptions.html]: Can you add a test that even if you modify the selection, e.g. by setting `select.selectedOptions[0].selected = false`, the object stays the same?

**Comment by Zirro** [comment]
Done. I replaced some of the .item() checks, but decided to keep a few as well in order to test that the value is/behaves like a HTMLCollection.

**Pull request merged**